eva_sub_cli/validators/validator.py

```diff
@@ -161,63 +164,96 @@ def check_if_file_missing(self):
-        self.results['vcf_check']['pass'] = vcf_check_result
-
-        # assembly_check result
-        asm_nb_mismatch_result = all((asm_check.get('nb_mismatch', 1) == 0
-                                      for vcf_name, asm_check in self.results.get('assembly_check', {}).items()))
-        asm_nb_error_result = all((asm_check.get('nb_error', 1) == 0
-                                   for vcf_name, asm_check in self.results.get('assembly_check', {}).items()))
-        self.results['assembly_check']['pass'] = asm_nb_mismatch_result and asm_nb_error_result
-
-        # fasta_check result
-        fasta_check_result = all((fa_file_check.get('all_insdc', False) is True
-                                  for fa_file, fa_file_check in self.results.get('fasta_check', {}).items()))
-        self.results['fasta_check']['pass'] = fasta_check_result
-
-        # sample check result
-        self.results['sample_check']['pass'] = self.results.get('sample_check', {}).get('overall_differences',
-                                                                                        True) is False
-
-        # evidence type check result
-        self.results['evidence_type_check']['pass'] = (bool(self.results.get('evidence_type_check')) and
-                                                       all('evidence_type' in v and v['evidence_type'] is not None
-                                                      for k, v in self.results.get('evidence_type_check', {}).items()
-                                                      if isinstance(v, dict)))
-
-        # metadata check result
-        metadata_xlsx_result = len(self.results.get('metadata_check', {}).get('spreadsheet_errors', []) or []) == 0
-        metadata_json_result = len(self.results.get('metadata_check', {}).get('json_errors', []) or []) == 0
-        self.results['metadata_check']['pass'] = metadata_xlsx_result and metadata_json_result
+
+        if VCF_CHECK in self.tasks:
+            # vcf_check result
+            vcf_check_result = all((vcf_check.get('critical_count', 1) == 0
+                                    for vcf_name, vcf_check in self.results.get('vcf_check', {}).items()))
+            self.results['vcf_check']['pass'] = vcf_check_result
+
+            # evidence type check result
+            self.results['evidence_type_check']['pass'] = (bool(self.results.get('evidence_type_check')) and
+                                                           all('evidence_type' in v and v['evidence_type'] is not None
+                                                               for k, v in
+                                                               self.results.get('evidence_type_check', {}).items()
+                                                               if isinstance(v, dict)))
+        elif VCF_CHECK not in self.previous_validation_results:
+            self.results.update({'vcf_check': {'pass': PROCESS_NOT_RUN_YET}})
```

Comment:
No reason to use `update` here and elsewhere in this method:
```suggestion
            self.results['vcf_check'] = {'pass': PROCESS_NOT_RUN_YET}}
```

Reply:
updated